openfisca_france/model/prestations/autonomie.py

```diff
@@ -0,0 +1,309 @@
+    entity = Individu
+    definition_period = MONTH
+
+
+    def formula_2002(individu, period, legislation):
+        # Les départements doivent appliquer la nouvelle formule
+        # entre le 1er mars 2016 et le 28 février 2017
+        base_ressources_apa = individu('base_ressources_apa', period)
+        en_couple = individu.famille('en_couple', period)
+        legislation = legislation(period.start).autonomie
+        seuil_inf = legislation.apa_domicile.seuil_de_revenu_en_part_du_mtp.seuil_inferieur
+        seuil_sup = legislation.apa_domicile.seuil_de_revenu_en_part_du_mtp.seuil_superieur
+        majoration_tierce_personne = legislation.mtp.mtp
+        taux_min_participation = legislation.apa_domicile.taux_de_participation_minimum
+        taux_max_participation = legislation.apa_domicile.taux_de_participation_maximum
+        proratisation_couple = (
+            1 +
+            en_couple * (legislation.apa_domicile.divison_des_ressources_du_menage_pour_les_couples - 1)
+            )
+        dependance_plan_aide_domicile_accepte = individu('dependance_plan_aide_domicile_accepte', period)
+        base_ressources_apa_domicile = base_ressources_apa / proratisation_couple
+
+        condition_ressources_domicile = [
+            base_ressources_apa_domicile <= (seuil_inf * majoration_tierce_personne),
+            (seuil_inf * majoration_tierce_personne) < base_ressources_apa_domicile <= (seuil_sup * majoration_tierce_personne),
+            base_ressources_apa_domicile > (seuil_sup * majoration_tierce_personne),
+            ]
+        taux_participation = [
+            taux_min_participation,
+            (base_ressources_apa_domicile - seuil_inf * majoration_tierce_personne) / ((seuil_sup - seuil_inf) * majoration_tierce_personne) * taux_max_participation,
+            taux_max_participation,
+            ]
+        apa_domicile_participation = select(condition_ressources_domicile, taux_participation) * dependance_plan_aide_domicile_accepte
+        return apa_domicile_participation
+
+    def formula_2016_03_01(individu, period, legislation):
+        # Les départements doivent appliquer la nouvelle formule
+        # entre le 1er mars 2016 et le 28 février 2017
+        base_ressources_apa = individu('base_ressources_apa', period)
+        en_couple = individu.famille('en_couple', period)
+        dependance_plan_aide_domicile_accepte = individu('dependance_plan_aide_domicile_accepte', period)
+        legislation = legislation(period.start).autonomie
+        majoration_tierce_personne = legislation.mtp.mtp
+        proratisation_couple = (
+            1 +
+            en_couple * (legislation.apa_domicile.divison_des_ressources_du_menage_pour_les_couples - 1)
+            )
+        base_ressources_apa_domicile = base_ressources_apa / proratisation_couple
+        # TODO: use a marignal tax scale
+        condlist = [
+            dependance_plan_aide_domicile_accepte <= (0.317 * majoration_tierce_personne),
```

Comment:
Idéalement ces paramètres en dur devraient être fourni pas les paramètres législatifs.
On doit avoir cela quelque part dans les barèmes IPP

Reply:
Do we have a legislative reference for this parameter ? In all case, it would be better to make it a parameter. Same for all the other hard-coded numbers.

Reply:
Les paramètres ayant été écrits "en dur" dans le code sont ceux qui ne seront amenés à être modifiés qu'avec une réforme structurelle de l'APA (qui nécessitera donc nécessairement de revérifier la totalité des formules). C'est pour cette raison que j'avais privilégié de laisser le paramètre sous cette forme.